providers/dns/rfc2136/rfc2136.go

```diff
@@ -48,27 +48,29 @@ func NewDNSProvider(nameserver, tsigAlgorithm, tsigKey, tsigSecret string) (*DNS
 }
 
 // Present creates a TXT record using the specified parameters
-func (r *DNSProvider) Present(domain, token, keyAuth string) error {
-	fqdn, value, ttl := acme.DNS01Record(domain, keyAuth)
-	return r.changeRecord("INSERT", fqdn, value, ttl)
+func (r *DNSProvider) Present(domain *acme.Domain, token, keyAuth string) error {
+	fqdn, value, ttl := domain.GetDNS01Record(keyAuth)
+	txtRecord := acme.NewDomain(fqdn)
```

Comment:
you know, you could have done the fqdn > acme.Domain conversion in changeRecord() and saved yourself a bit of work ^^